src/checker.rs

```diff
@@ -479,16 +527,21 @@ impl ast::UnscopedVariable {
     }
 
     fn check_get(&mut self, ctx: &mut CheckContext) -> Result<ExpressionResult, CheckError> {
-        // If the variable is not found, we return a default value for a possible global variable.
-        let value = ctx
-            .locals
-            .get(&self.name)
-            .cloned()
-            .unwrap_or_else(|| ExpressionResult {
+        if let Some(result) = ctx.globals.get(&self.name) {
+            Some(result)
+        } else {
+            ctx.locals.get(&self.name)
+        }
+        .cloned()
+        // TODO deprecated support for undeclared globals
```

Comment:
Do we need to keep this?  @robrix @tausbn, I think you're the primary current users of this, would it be problematic to go ahead and make undeclared globals a hard error now?